Automattic/jetpack
# CRM: Client Portal custom fields not working

## Proposed changes:

* This PR fixes the following problems in the Client Portal
  * Numeric Fields (this includes Numeric with decimals) - numeric fields were not being rendered
  * TextAreas - Special characters in textareas were not being displayed correctly. This could potentially insert wrong data in the database (characters that were wrongly inserted by the escape function used)
  * Date Fields - Date fields were not working correctly

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to Settings -> Custom Fields (`wp-admin/admin.php?page=zerobscrm-plugin-settings&tab=customfields`) and add the following for Contacts:
  * Numeric Field
  * Numeric (Decimals) Field
  * Textarea
  * Date
* Log in the Client Portal (`/clients`) as a Contact (if you don't have one yet, you can add the client portal access to contacts in their edit page `admin.php?page=zbs-add-edit&action=view&zbstype=contact&zbsid=1`)
* Change the information for each field, making sure to add special characters to the textarea (e.g. `<some spéciál $# characters>`

In `trunk` numeric fields don't get rendered, textareas will fail to escape special characters and date fields won't work.

In `crm/fix/client_portal_custom_fields_not_working` everything should work as expected.
